File: BaekJoon/투포인터/1806_부분 합.kt
import java.util.*

fun main() {
    val sc = Scanner(System.`in`)
    val n = sc.nextInt()
    val s = sc.nextInt()
    val array = IntArray(n) { sc.nextInt() }
    var partSum = 0
    var partLength = 100000
    var left = 0
    var right = 0

    while (true) {
        if (partSum >= s) {
            partLength = Math.min(partLength, right - left)
            partSum -= array[left++]
        } else if (right == n) {
            break
        } else {
            partSum += array[right++]
        }
    }

    if (partLength == 100000) print(0)
    else print(partLength)
}

File: BaekJoon/투포인터/1806_부분합.swift
/*
10 15
5 1 3 5 10 7 4 9 2 8
*/

let info = readLine()!.split(separator: " ").map { Int(String($0))! }
let numbers = readLine()!.split(separator: " ").map { Int(String($0))! }

var left = 0
var right = 0
var minLength = 100_000
var partSum = 0

while left <= right {
    if partSum >= info[1] {
        minLength = min(minLength, right - left)

        partSum -= numbers[left]
        left += 1
    } else if (right == info[0]) {
        break
    } else {
        partSum += numbers[right]
        right += 1
    }
}

print(minLength ==  100_000 ? 0 : minLength)